prime-router/src/main/kotlin/cli/helpers/HL7DiffHelper.kt

```diff
@@ -336,4 +346,27 @@ class HL7DiffHelper {
                 " Differences: $input$outputText"
         }
     }
+
+    /**
+     * helper
+     * Check that v1 and v2 are effectively both blank value
+     */
+    fun effectivelyBlank(v1: Array<ca.uhn.hl7v2.model.Type>, v2: Array<ca.uhn.hl7v2.model.Type>): Boolean {
+        return (
+            v1.size == 1 &&
+            v1[0].isEmpty &&
+            v2.isEmpty()
+        )
+    }
+
+    /**
+     * helper - heuristically check input and output are compatible types
+     */
+    private fun typeCompatible(t1: Any, t2: Any): Boolean {
+        return (t1.javaClass == t2.javaClass) ||
+                (
+                    t1 is ca.uhn.hl7v2.model.AbstractPrimitive &&
+                t2 is ca.uhn.hl7v2.model.AbstractPrimitive
+                )
```

Comment:
(nit) I think this formatting is cleaner
```suggestion
        return (t1.javaClass == t2.javaClass) ||
            (t1 is ca.uhn.hl7v2.model.AbstractPrimitive && t2 is ca.uhn.hl7v2.model.AbstractPrimitive)
```